hooks/mari/publish_mipmaps.py

```diff
@@ -0,0 +1,457 @@
+                if seq_path:
+                    src_files = publisher.util.get_sequence_path_files(source_path)
+                else:
+                    src_files = [source_path]
+
+                # write mipmaps for each file in the sequence
+                item.properties["mipmap_paths"].extend(self.create_mipmaps_for_seq(src_files, target_path))
+
+            except Exception as e:
+                raise TankError("Failed to publish file for item '%s': %s" % (item.name, str(e)))
+
+            self.logger.debug(
+                "Published file for item '%s' to '%s'." % (item.name, target_path)
+            )
+
+        return item.properties["mipmap_paths"]
+
+    def publish(self, task_settings, item):
+        """
+        Executes the publish logic for the given item and task_settings.
+
+        :param task_settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the task_settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+        """
+        # TODO: check if Publish Textures plugin was successfully run for the same item?
+        publisher = self.parent
+
+        # Get item properties populated by validate method
+        publish_type          = item.properties.publish_type
+        publish_version       = item.properties.publish_version
+        publish_user          = item.get_property("publish_user", default_value=None)
+
+        # handle publishing of files first
+        self.publish_files(task_settings, item)
+
+        # Get any upstream dependency paths
+        dependency_paths = item.properties.get("publish_dependency_paths", [])
+
+        # Find upstream dependency from current item.properties.sg_publish_data_list
+        # since product of publish_texture is the upstream dependency, and the item
+        # is shared between the plugins
+        dependency_ids = [x["id"] for x in item.properties.get("sg_publish_data_list")]
+        # Now reset publish_data_list
+        item.properties["sg_publish_data_list"] = []
+
+        # If the parent item has publish data, include those ids in the
+        # list of dependencies as well
+        if "sg_publish_data_list" in item.parent.properties:
+            dependency_ids.extend([ent["id"] for ent in item.parent.properties["sg_publish_data_list"]])
+
+        # get any additional_publish_fields that have been defined
+        sg_fields = {}
+        additional_fields = task_settings.get("additional_publish_fields").value or {}
+        for template_key, sg_field in additional_fields.iteritems():
+            if template_key in item.properties.fields:
+                sg_fields[sg_field] = item.properties.fields[template_key]
+
+
+        # Make sure any specified fields exist on the PublishedFile entity
+        sg_fields = self._validate_sg_fields(sg_fields)
+
+        for extension in item.properties.mipmap_extensions:
```

Comment:
I'm assuming the necessity for overwriting this method goes away with my suggested implementation of using multiple mipmap plugin instances rather than a list of extensions. Please let me know though if I'm wrong...its a bummer to have to overwrite such a base level method.